src/expr.cpp

```diff
@@ -2034,12 +2034,21 @@ static ConstExpr *lConstFoldBinaryIntOp(BinaryExpr::Op op, const T *v0, const T
     int count = carg0->Count();
 
     switch (op) {
-        FOLD_OP_REF(BinaryExpr::Mod, %, TRef);
         FOLD_OP_REF(BinaryExpr::Shl, <<, TRef);
         FOLD_OP_REF(BinaryExpr::Shr, >>, TRef);
         FOLD_OP_REF(BinaryExpr::BitAnd, &, TRef);
         FOLD_OP_REF(BinaryExpr::BitXor, ^, TRef);
         FOLD_OP_REF(BinaryExpr::BitOr, |, TRef);
+    case BinaryExpr::Mod:
+        for (int i = 0; i < count; ++i) {
+            if (v1[i] == 0) {
+                Warning(pos, "Division by zero encountered in expression.");
```

Comment:
It's not division, clang gives the following message: "warning: remainder by zero is undefined".
For division it has: "division by zero is undefined".

Reply:
Whoops..will change it - I checked with gcc : https://godbolt.org/z/WYMXnn .
Clang was different - https://godbolt.org/z/D9M9tw

Reply:
Done